packages/core/src/extensions/SideMenu/SideMenuPlugin.ts

```diff
@@ -706,6 +709,23 @@ export class SideMenuProsemirrorPlugin<
           });
           return this.view;
         },
+        props: {
+          handleDrop: (_view, _event, slice) => {
+            const draggedBlock = nodeToBlock(
+              slice.content.child(0),
+              editor.pmSchema,
+            );
+            this.draggedBlockId = draggedBlock.id;
+            return false;
+          },
+        },
+        filterTransaction: (tr) => {
+          if (tr.getMeta("uiEvent") === "drop" && this.draggedBlockId) {
+            tr.setMeta("draggedBlockId", this.draggedBlockId);
+            this.draggedBlockId = undefined;
+          }
+          return true;
+        },
```

Comment:
Ended up attaching the `draggedBlockId` to the transaction, I'm unsure if this is the ideal approach though 😅 

https://github.com/user-attachments/assets/e37787fb-08d7-4b6b-8ff8-993b847f5112

Reply:
Hm, yea. This is a good argument for over reporting on all of the indexes that have changed.

I don't want for dragged blocks to be handled specially in some sort of way, since that will mean having to maintain every possible way the editor may change (to get the most "fine-grained" update), when this should be a general solution.

Sorry to make you do more work here. I didn't see your comment since it was marked as resolved.

Reply:
That sounds like it might just work!

Reply:
@rmolinamir I had a go at this with https://github.com/TypeCellOS/BlockNote/pull/1960 and I think I've got something which should work well enough. Appreciate you bring it up & working on this

Reply:
Thanks for the good work! I'll keep an eye out for anything else and contribute. Love this library ❤️